lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -528,15 +528,25 @@ void HdVP2Mesh::_PrepareSharedVertexBuffers(
             if (computeCPUNormals) {
                 // note: normals gets dirty when points are marked as dirty,
                 // at change tracker.
-                Hd_VertexAdjacencySharedPtr adjacency(new Hd_VertexAdjacency());
-                HdBufferSourceSharedPtr     adjacencyComputation
-                    = adjacency->GetSharedAdjacencyBuilderComputation(&_meshSharedData->_topology);
-                adjacencyComputation->Resolve();
+                if (!_meshSharedData->_adjacency) {
+                    _meshSharedData->_adjacency.reset(new Hd_VertexAdjacency());
+
+                    HdBufferSourceSharedPtr adjacencyComputation
+                        = _meshSharedData->_adjacency->GetSharedAdjacencyBuilderComputation(
+                            &_meshSharedData->_topology);
+                    MProfilingScope profilingScope(
+                        HdVP2RenderDelegate::sProfilerCategory,
+                        MProfiler::kColorC_L2,
+                        _rprimId.asChar(),
+                        "HdVP2Mesh::computeAdjacency");
+
+                    adjacencyComputation->Resolve();
```

Comment:
This call to resolve is slow, so save the result and only re-do it if the topology actually changes.